lib/livebook_web/live/path_select_component.ex

```diff
@@ -49,17 +49,31 @@ defmodule LivebookWeb.PathSelectComponent do
           </div>
         <% end %>
       </div>
-      <div class="flex-grow -m-1 p-1 overflow-y-auto tiny-scrollbar">
-        <div class="grid grid-cols-4 gap-2">
-          <%= for file <- list_matching_files(@path, @extnames, @running_paths) do %>
-            <%= render_file(file, @phx_target) %>
-          <% end %>
-        </div>
+      <div class="flex-grow -m-1 p-1 overflow-y-auto tiny-scrollbar divide-y divide-grey-200 divide-dashed">
+        <%= for file_group <- file_groups(@path, @extnames, @running_paths) do %>
+          <div class="grid grid-cols-4 gap-2">
+            <%= for file <- file_group do %>
+              <%= render_file(file, @phx_target) %>
+            <% end %>
+          </div>
+        <% end %>
       </div>
     </div>
     """
   end
 
+  defp file_groups(path, extnames, running_paths) do
+    files = list_matching_files(path, extnames, running_paths)
+
+    List.delete(
+      [
+        Enum.filter(files, fn file -> file.highlighted != "" end),
+        Enum.filter(files, fn file -> file.highlighted == "" end)
+      ],
+      []
+    )
```

Comment:
There is a `Enum.split_with` function you can use instead. :)

Reply:
Apologies, for the ignorance. (I really should be looking at the docs more carefully!)

Reply:
Oh, it is no problem at all, the Enum module is has a large surface API anyway, so it takes some time to get used to!